server/src/test/java/org/elasticsearch/snapshots/RegisteredPolicySnapshotsSerializationTests.java

```diff
@@ -0,0 +1,83 @@
+
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.test.AbstractChunkedSerializingTestCase;
+import org.elasticsearch.test.ESTestCase;
+import org.elasticsearch.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class RegisteredPolicySnapshotsSerializationTests extends AbstractChunkedSerializingTestCase<RegisteredPolicySnapshots> {
+    @Override
+    protected RegisteredPolicySnapshots doParseInstance(XContentParser parser) throws IOException {
+        return RegisteredPolicySnapshots.parse(parser);
+    }
+
+    @Override
+    protected Writeable.Reader<RegisteredPolicySnapshots> instanceReader() {
+        return RegisteredPolicySnapshots::new;
+    }
+
+    @Override
+    protected RegisteredPolicySnapshots createTestInstance() {
+        return randomRegisteredPolicySnapshots();
+    }
+
+    @Override
+    protected RegisteredPolicySnapshots mutateInstance(RegisteredPolicySnapshots instance) throws IOException {
+        if (instance.getSnapshots().isEmpty()) {
+            return new RegisteredPolicySnapshots(List.of(randomPolicySnapshot()));
+        }
+
+        final int randIndex = between(0, instance.getSnapshots().size() - 1);
+        final RegisteredPolicySnapshots.PolicySnapshot policySnapshot = instance.getSnapshots().get(randIndex);
+
+        String policy = policySnapshot.getPolicy();
+        String snapshotName = policySnapshot.getSnapshotId().getName();
+        String snapshotUUID = policySnapshot.getSnapshotId().getUUID();
+
+        switch (between(0, 2)) {
+            case 0 -> {
+                policy = randomPolicy();
+            }
+            case 1 -> {
+                snapshotName = randomValueOtherThan(snapshotName, ESTestCase::randomIdentifier);
+            }
+            case 2 -> {
+                snapshotUUID = randomValueOtherThan(snapshotName, ESTestCase::randomUUID);
+            }
+            default -> throw new AssertionError("failure, got illegal switch case");
+        }
+
+        List<RegisteredPolicySnapshots.PolicySnapshot> newSnapshots = new ArrayList<>(instance.getSnapshots());
+        newSnapshots.set(randIndex, new RegisteredPolicySnapshots.PolicySnapshot(policy, new SnapshotId(snapshotName, snapshotUUID)));
+        return new RegisteredPolicySnapshots(newSnapshots);
+    }
+
+    private RegisteredPolicySnapshots randomRegisteredPolicySnapshots() {
+        final List<RegisteredPolicySnapshots.PolicySnapshot> snapshots = new ArrayList<>();
+        for (int i = 0; i < randomIntBetween(0, 20); i++) {
+            snapshots.add(randomPolicySnapshot());
+        }
+        return new RegisteredPolicySnapshots(snapshots);
+    }
+
+    private String randomPolicy() {
+        return "policy-" + randomIntBetween(0, 20);
+    }
+
+    private RegisteredPolicySnapshots.PolicySnapshot randomPolicySnapshot() {
+        SnapshotId snapshotId = new SnapshotId(randomIdentifier(), randomUUID());
```

Comment:
It shouldn't be possible in the real code for UUID to be null here. It should serialize fine, but this class is only meant to test valid  objects, right?

Reply:
I think it could be `null` if sent to a newer node from a pre-this-change master node?

Reply:
I believe in that case the call to `request.uuid()` at https://github.com/elastic/elasticsearch/pull/110216/files#diff-de99cbad875eb82467666e03b94c5b9376e92ce73fa021a99ab58ea4c2bae4f6R280 should have set the uuid on the request before it is then added to the registered snapshots at https://github.com/elastic/elasticsearch/pull/110216/files#diff-de99cbad875eb82467666e03b94c5b9376e92ce73fa021a99ab58ea4c2bae4f6R3963

Am I not considering a way this could happen?

Also, looks like uuid can't be null in SnapshotId: https://github.com/elastic/elasticsearch/blob/c7819aa91adb41fac547c406d1d1d83de3dae5d4/server/src/main/java/org/elasticsearch/snapshots/SnapshotId.java#L56